data-donation-sdk/src/commonMain/kotlin/care/data4life/datadonation/donation/program/model/ProgramModel.kt

```diff
@@ -31,52 +31,60 @@ enum class BlurFunction(val value: String) {
 }
 
 @Serializable(with = BlurFunctionSerializer::class)
-enum class RevocationMode(val value: String) {
+internal enum class RevocationMode(val value: String) {
     DELETE("delete"),
     ANONYMIZE("anonymize")
 }
 
 @Serializable
-internal data class ProgramResourceBlurItem(
+internal data class QuestionnaireResponseItemBlur(
     val linkId: String,
     @SerialName("fn")
     @Contextual
     val function: BlurFunction
 )
 
 @Serializable
-internal data class ProgramResourceBlur(
-    val location: String? = null,
+internal data class ProgramFhirResourceBlur(
+    @SerialName("location")
+    val targetTimeZone: String? = null,
     @Contextual
-    val authored: BlurFunction? = null,
-    val items: List<ProgramResourceBlurItem>
+    @SerialName("authored")
+    val questionnaireResponseAuthored: BlurFunction? = null,
+    @SerialName("items")
+    val itemBlurs: List<QuestionnaireResponseItemBlur>
 )
 
 @Serializable
-internal data class ProgramResource(
+internal data class ProgramFhirResourceConfiguration(
     val url: String,
     val versions: List<String>? = null,
-    val blur: ProgramResourceBlur? = null
+    @SerialName("blur")
+    val fhirBlur: ProgramFhirResourceBlur? = null
 )
 
 @Serializable
-internal data class ProgramAnonymizationBlur(
-    val location: String,
+internal data class ProgramAnonymizationGlobalBlur(
+    @SerialName("location")
+    val targetTimeZone: String,
     @Contextual
-    val authored: BlurFunction? = null,
+    @SerialName("authored")
+    val questionnaireResponseAuthored: BlurFunction? = null,
     @Contextual
     val researchSubject: BlurFunction? = null
 )
 
 @Serializable
 internal data class ProgramAnonymization(
-    val blur: ProgramAnonymizationBlur? = null
+    @SerialName("blur")
+    val globalBlur: ProgramAnonymizationGlobalBlur? = null
 )
```

Comment:
cant we skip this class and just add the inner optional blur rule instead? not sure what ktor allows

Reply:
We need a separate class for this unless the backend gets rid of this overdo.

Reply:
no i mean, can the serializer skip it? i dont know map  it directly in the class containing this?

Reply:
No and it would need a lot of additional code to make it happen, since we use the build-in Serializers, which are automatically generated (aside form Enums).